packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -226,485 +226,414 @@ export interface CodeEditorProps extends Omit<React.HTMLProps<HTMLDivElement>, '
-      return {
-        value: nextProps.code,
-        prevPropsCode: nextProps.code
-      };
-    }
-    // else, don't need to change the state.value
-    // because the onChange function will do all the work
-    return null;
-  }
-
-  handleResize = () => {
-    if (this.editor) {
-      this.editor.layout({ width: 0, height: 0 }); // ensures the editor won't take up more space than it needs
-      this.editor.layout();
+  const handleResize = () => {
+    if (editorRef.current) {
+      editorRef.current.layout({ width: 0, height: 0 }); // ensures the editor won't take up more space than it needs
+      editorRef.current.layout();
     }
   };
 
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleGlobalKeys);
-    this.observer = getResizeObserver(this.ref.current, this.handleResize, true);
-    this.handleResize();
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleGlobalKeys);
-    this.observer();
-  }
-
-  handleGlobalKeys = (event: KeyboardEvent) => {
-    if (this.wrapperRef.current === document.activeElement && (event.key === 'ArrowDown' || event.key === ' ')) {
-      this.editor?.focus();
+  const handleGlobalKeys = (event: KeyboardEvent) => {
+    if (wrapperRef.current === document.activeElement && (event.key === 'ArrowDown' || event.key === ' ')) {
+      editorRef.current?.focus();
       event.preventDefault();
     }
   };
 
-  editorDidMount: EditorDidMount = (editor, monaco) => {
+  // if the code changes due to the props.code changing
+  // set the value to props.code
+  useEffect(() => setValue(props.code), [props.code]);
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleGlobalKeys);
+    observer.current = getResizeObserver(ref.current, handleResize, true);
+    handleResize();
+    return () => {
+      document.removeEventListener('keydown', handleGlobalKeys);
+      observer.current();
+    };
+  }, []);
+
+  const editorDidMount: EditorDidMount = (editor, monaco) => {
     // eslint-disable-next-line no-bitwise
-    editor.addCommand(monaco.KeyMod.Shift | monaco.KeyCode.Tab, () => this.wrapperRef.current.focus());
+    editor.addCommand(monaco.KeyMod.Shift | monaco.KeyCode.Tab, () => wrapperRef.current.focus());
     Array.from(document.getElementsByClassName('monaco-editor')).forEach((editorElement) =>
       editorElement.removeAttribute('role')
     );
-    this.props.onEditorDidMount(editor, monaco);
-    this.editor = editor;
-    if (this.props.height === 'sizeToFit') {
-      this.setHeightToFitContent();
+    props.onEditorDidMount(editor, monaco);
+    editorRef.current = editor;
+    if (props.height === 'sizeToFit') {
+      setHeightToFitContent();
     }
   };
 
-  handleFileChange = (value: string, filename: string) => {
-    this.setState({
-      value,
-      filename
-    });
-    this.props.onCodeChange(value);
+  const handleFileChange = (value: string) => {
```

Comment:
@nicolethoen do you recall the original reason for having a `filename` state? Didn't seem like it was being used anywhere other than being set.